pandas/io/excel.py

```diff
@@ -1748,6 +1754,9 @@ def write_cells(self, cells, sheet_name=None, startrow=0, startcol=0,
                 num_format_str = self.datetime_format
             elif isinstance(cell.val, date):
                 num_format_str = self.date_format
+            elif isinstance(cell.val, timedelta):
```

Comment:
any way to make a function to do all of these conversions that will work across engines?

Reply:
Fair callout - the latest commit should account for that.

FWIW I didn't want to do it here but there appears to be a good refactoring opportunity for `write_cells` (in fact the xlwt and xlsxwriter implementations of this are *almost* identical).

Rather than making `write_cells` an abstract method I think it's possible to implement that in the base class with the following pseudo-code:

```python
get_or_add_sheet_name
handle_freeze_panes
loop_over_cells:
    format_cell
    handle_cell_merges
    write_cell
```

If we did it that way and made the steps above abstract methods there would be much less copy/paste than what's in the module today. Again didn't want to do it here but happy to open as a new issue if you think its worth it